firmware/src/network_model.c

```diff
@@ -0,0 +1,837 @@
+    return RDMResponder_BuildNack(header, NR_FORMAT_ERROR);
+  }
+
+  unsigned int id = JoinUInt32(param_data[0], param_data[1],
+                               param_data[2], param_data[3]);
+  int index = LookupIndex(id);
+  if (index < 0) {
+    return RDMResponder_BuildNack(header, NR_DATA_OUT_OF_RANGE);
+  }
+
+  if (index == IPSEC_INTERFACE_ID) {
+    return RDMResponder_BuildNack(header, NR_ACTION_NOT_SUPPORTED);
+  }
+
+  g_network_model.interfaces[index].configured_dhcp_mode = param_data[4];
+
+  return RDMResponder_BuildSetAck(header);
+}
+
+int NetworkModel_GetZeroconfMode(const RDMHeader *header,
+                                 const uint8_t *param_data) {
+  unsigned int index = 0;
+  int r = CheckInterfaceRequest(header, param_data, &index);
+  if (r >= 0) {
+    return r;
+  }
+
+  ReturnUnlessUnicast(header);
+
+  unsigned int offset = sizeof(RDMHeader);
+  memcpy(g_rdm_buffer + offset, param_data, INTERFACE_ID_SIZE);
+  offset += INTERFACE_ID_SIZE;
+  g_rdm_buffer[offset++] =
+      g_network_model.interfaces[index].configured_zeroconf_mode;
+  RDMResponder_BuildHeader(header, ACK, GET_COMMAND_RESPONSE,
+                           ntohs(header->param_id),
+                           offset - sizeof(RDMHeader));
+  return RDMUtil_AppendChecksum(g_rdm_buffer);
+}
+
+int NetworkModel_SetZeroconfMode(const RDMHeader *header,
+                                 const uint8_t *param_data) {
+  if (header->param_data_length != sizeof(uint32_t) + sizeof(uint8_t)) {
+    return RDMResponder_BuildNack(header, NR_FORMAT_ERROR);
+  }
+
+  unsigned int id = JoinUInt32(param_data[0], param_data[1],
+                               param_data[2], param_data[3]);
+  int index = LookupIndex(id);
+  if (index < 0) {
+    return RDMResponder_BuildNack(header, NR_DATA_OUT_OF_RANGE);
+  }
+
+  if (index == IPSEC_INTERFACE_ID) {
+    return RDMResponder_BuildNack(header, NR_ACTION_NOT_SUPPORTED);
+  }
+
+  g_network_model.interfaces[index].configured_zeroconf_mode = param_data[4];
+
+  return RDMResponder_BuildSetAck(header);
+}
+
+int NetworkModel_GetCurrentAddress(const RDMHeader *header,
+                                   const uint8_t *param_data) {
+  unsigned int index = 0;
+  int r = CheckInterfaceRequest(header, param_data, &index);
+  if (r >= 0) {
+    return r;
+  }
+
+  ReturnUnlessUnicast(header);
+  InterfaceState *interface = &g_network_model.interfaces[index];
+
+  unsigned int offset = sizeof(RDMHeader);
+  memcpy(g_rdm_buffer + offset, param_data, INTERFACE_ID_SIZE);
+  offset += INTERFACE_ID_SIZE;
+  g_rdm_buffer[offset++] = UInt32Byte0(interface->current_ip);
+  g_rdm_buffer[offset++] = UInt32Byte1(interface->current_ip);
+  g_rdm_buffer[offset++] = UInt32Byte2(interface->current_ip);
+  g_rdm_buffer[offset++] = UInt32Byte3(interface->current_ip);
+  g_rdm_buffer[offset++] = interface->current_netmask;
+  if (index == IPSEC_INTERFACE_ID) {
+    g_rdm_buffer[offset++] = DHCP_MODE_UNKNOWN;
```

Comment:
The new text is:

"Some devices may not be able to determine if the current address was obtained via DHCP. If the device does not support this feature, the value of the DHCP Status field shall be DHCP_STATUS_UNKNOWN."

I can see how this can be read both ways. It's not clear if 'this feature' means DHCP or the ability to determine if the address was assigned via DHCP. 

I'll add this to my notes.